modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Blogs/MongoBlogPostRepository.cs

```diff
@@ -118,6 +110,28 @@ public class MongoBlogPostRepository : MongoDbRepository<CmsKitMongoDbContext, B
                                     }).ToList();
     }
 
+    protected virtual async Task<List<Guid>> GetEntityIdsByTagId(Guid? tagId, CancellationToken cancellationToken)
+    {
+        var entityIdFilters = new List<Guid>();
+        if (!tagId.HasValue)
+        {
+            return entityIdFilters;
+        }
+
+        var entityIds =
+            await _entityTagManager.GetEntityIdsFilteredByTagAsync(tagId.Value, CurrentTenant.Id, cancellationToken);
+            
+        foreach (var entityId in entityIds)
+        {
+            if (Guid.TryParse(entityId, out var guid))
```

Comment:
I think 'guid' is not a good naming here, instead you can rename it like `entityIdAsGuid`, `parsedEntityId`, or something else.